src/main/java/org/testng/TestNG.java

```diff
@@ -1490,7 +1490,7 @@ public static TestNG privateMain(String[] argv, ITestListener listener) {
       else {
         error(ex.getMessage());
       }
-      result.setStatus(HAS_FAILURE);
+      result.exitCode = ExitCode.newExitCodeRepresentingFailure();
```

Comment:
Why not `result.exitCode.setFailure()`?

Reply:
I don't think it makes sense to expose setters into `ExitCode`. What would be wrong in having a factory method produce `ExitCode` instance that represents failure ?

Reply:
ok